src/PHPMailer.php

```diff
@@ -3897,35 +3898,46 @@ public function msgHTML($message, $basedir = '', $advanced = false)
                     );
                     continue;
                 }
-                if (// Only process relative URLs if a basedir is provided (i.e. no absolute local paths)
-                    !empty($basedir)
-                    // Ignore URLs containing parent dir traversal (..)
-                    and (strpos($url, '..') === false)
+                if (// Ignore URLs containing parent dir traversal (..)
+                    (strpos($url, '..') === false)
                     // Do not change urls that are already inline images
                     and 0 !== strpos($url, 'cid:')
                     // Do not change absolute URLs, including anonymous protocol
                     and !preg_match('#^[a-z][a-z0-9+.-]*:?//#i', $url)
                 ) {
-                    $filename = static::mb_pathinfo($url, PATHINFO_BASENAME);
-                    $directory = dirname($url);
-                    if ('.' == $directory) {
-                        $directory = '';
-                    }
                     $cid = hash('sha256', $url) . '@phpmailer.0'; // RFC2392 S 2
-                    if (strlen($basedir) > 1 and '/' != substr($basedir, -1)) {
-                        $basedir .= '/';
-                    }
-                    if (strlen($directory) > 1 and '/' != substr($directory, -1)) {
-                        $directory .= '/';
+                    $added = false;
+
+                    // Only process relative URLs if a basedir is provided (i.e. no absolute local paths)
+                    if (is_string($imageResolver) && !empty($imageResolver)) {
+                        $filename = static::mb_pathinfo($url, PATHINFO_BASENAME);
+                        $directory = dirname($url);
+                        if ('.' == $directory) {
+                            $directory = '';
+                        }
+                        if (strlen($directory) > 1 and '/' != substr($directory, -1)) {
+                            $directory .= '/';
+                        }
+                        $added = $this->addEmbeddedImage(
+                            $imageResolver . $directory . $filename,
+                            $cid,
+                            $filename,
+                            static::ENCODING_BASE64,
+                            static::_mime_types((string) static::mb_pathinfo($filename, PATHINFO_EXTENSION))
+                        );
+                    } elseif (is_callable($imageResolver)) {
```

Comment:
If the passed param is not callable we can thrown an exception telling about it. Otherwise it will be hard to debug what exactly is happening